### PR TITLE
Made adapters in config an array. Sort user adapters based on internal adapters

### PR DESCRIPTION
**Pull request by maxrbaldwin**
1. Made adapters in config an array. Indexes can be an object or string.
2. Made `src/adapters/index.js` export all adapters as one object.
3. Wrote a util that returns an array of adapters based on user provided adapters.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
+    };
 
-  div.appendChild(reportButton);
-});
+    placement.appendChild(reportButton);
+  });
+  // run onload for adapters
+  adapters.forEach(adapter => {
+    if (adapter.onload) {
+      adapter.onload();
```
Comment (trinhathu) [inline src/index.js]: Can we also pass the config to each adapter's onload() call?

File: src/index.js [hunk 2]
```diff
@@ -1,18 +1,36 @@
...
 
-  reportButton.classList.add('report-ad-button');
-  reportButton.innerHTML = '<span>Report Ad!</span>';
-  reportButton.id = `report-button-${i}`;
-  reportButton.onclick = handleClick;
+    reportButton.classList.add('report-ad-button');
+    reportButton.innerHTML = '<span>Report Ad!</span>';
+    reportButton.id = `report-button-${i}`;
+    reportButton.onclick = function (e) {
+      handleclick(e, adapters, config);
```
Comment (trinhathu) [inline src/index.js]: Brian suggested earlier today that we remove handleclick() and just loop through the adapters here, like we do below for onload. Do you like that idea?

Comment (maxrbaldwin) [reply src/index.js]: yeah that is fine. We can abstract it later if we need to

Comment (maxrbaldwin) [reply src/index.js]: nvm, I still abstracted this.

File: src/handlers/onclick.js [hunk 3]
```diff
@@ -1,3 +1,11 @@
-export default function handleClick(e) {
-  console.log('click: ', e);
+export default function handleclick({ event, adapters, config }) {
+  const $buttonClicked = event.srcElement;
```
Comment (trinhathu) [inline src/handlers/onclick.js]: I think adapters are expecting the ad element, so `event.target.parentElement`, I think.

**Pull request merged**